subprojects/packagefiles/highs/check/meson.build

```diff
@@ -0,0 +1,77 @@
+
+catch_dep = declare_dependency(
+  include_directories : include_directories('../extern')
+)
```

Comment:
We should ignore the vendored code and make use of dependency('catch2'). This is already in the WrapDB. If you are interested in submitting meson support upstream, then they could vendor catch2.wrap as well.

Reply:
I have a problem here now because the version they vendor shadows the dependency. I need to include the `extern` include dir also for other headers needed by the source and this bleeds into the test build.
…especially inconvenient because their old version of catch2 does not build on my system.

Reply:
Those other bits of code in extern/ are also external projects -- you could try adding doctest and pdqsort to the wrapdb, but I have no idea what filereaderlp is, it's not googleable, and none of the files contain accreditation for the source location they are pulled from.

Reply:
I think filereaderlp is `https://github.com/feldmeier/FilereaderLP`.
But actually ipm, basislu and a bunch of other parts could also be interpreted as extern projects, but they are melded into the codebase with various kinds of glue. Maybe the vendored filereaderlp also has or will in the future have similar hacks in it.

As a third party build system, I would rather not dig deeply into these details. It would be easier to drop support for building the test suite. The current implementation is not complete anyway. It does not build the binary or the julia bindings and the tests associated with those.

Reply:
If the vendored filereaderlp becomes a fork, then it should be moved out of extern/ surely. I don't see this as a problem.

Thanks for tracking down the provenance of that code. It includes a pkg-config file so it seems designed to be a stable public interface ;) definitely makes sense to use it as a subproject.